### PR TITLE
internal/api: refactor the api with the certification DB

### PR DESCRIPTION
**Pull request by jmontesi**

### CODE REVIEW

File: cnf-certification-test/certification/suite.go [hunk 1]
```diff
@@ -43,51 +43,55 @@ const (
...
 	ginkgo.BeforeEach(func() {
+		var err error
 		env = provider.GetTestEnvironment()
-		err := api.LoadCatalog(env.GetOfflineDBPath())
+		certificator, err = certdb.GetCertificator(env.GetOfflineDBPath())
 		if err != nil {
-			logrus.Infof("Offline DB not available, err: %v", err)
+			errMsg := fmt.Sprintf("Cannot access the certification DB, err: %v", err)
+			logrus.Errorf(errMsg)
+			ginkgo.Fail(errMsg)
```
Comment (greyerof) [inline cnf-certification-test/certification/suite.go:54]: Maybe we already have this pattern somewhere else, but I don't think it's necessary to log the same msg with logrus and  with the ginkgo.Fail(). Both will appear in the stdout, so, in this cases, I think that only the ginkgo.Fail(errMsg) is needed.

File: cnf-certification-test/certification/certtool/certtool.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ const (
 )
 
 var (
-	CertAPIClient api.CertificationValidator
+	CertAPIClient certdb.ResourceCertificator
```
Comment (greyerof) [inline cnf-certification-test/certification/certtool/certtool.go:33]: I don't think this variable is needed anymore. Same with the functions that use it in this file: they seem not to be used:
GetContainerCertificationRequestFunction
GetOperatorCertificationRequestFunction
WaitForCertificationRequestToSuccess

File: internal/certdb/certdb.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import (
+	"fmt"
+
+	"github.com/test-network-function/cnf-certification-test/internal/certdb/offlinecheck"
+	"github.com/test-network-function/cnf-certification-test/internal/certdb/onlinecheck"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+//go:generate moq -out api_moq.go . CertAPIClientFuncs
+type ResourceCertificator interface {
```
Comment (greyerof) [inline internal/certdb/certdb.go:12]: This is a personal opinion, but "Certificator" suffix seems a bit misleading to me, as we're not certifiying any resources, just getting their certification status. I would stick to "StatusChecker" or "CertificationStatusValidator". I see you also changed some function receiver names from "checker" to "validator", so maybe the second option is better for consistency.

File: internal/certdb/certdb.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"fmt"
+
+	"github.com/test-network-function/cnf-certification-test/internal/certdb/offlinecheck"
+	"github.com/test-network-function/cnf-certification-test/internal/certdb/onlinecheck"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+//go:generate moq -out api_moq.go . CertAPIClientFuncs
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply internal/certdb/certdb.go:11]: Done.

File: internal/certdb/certdb.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+func GetCertificator(offlineDBPath string) (ResourceCertificator, error) {
+	// use the online certificator by default
+	onlineValidator := onlinecheck.NewOnlineValidator()
+	if onlineValidator.IsServiceReachable() {
+		return onlineValidator, nil
+	}
+
+	// use the offline DB for disconnected environments
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply internal/certdb/certdb.go:25]: Done.

File: cnf-certification-test/certification/suite.go (hunk 1 above)
Comment (jmontesi) [reply cnf-certification-test/certification/suite.go:54]: Done.

**Review by greyerof — APPROVED** [review]

**Pull request merged**